homeassistant/components/lightwave/climate.py

```diff
@@ -0,0 +1,164 @@
+    """Find and return LightWave lights."""
+    if not discovery_info:
+        return
+
+    trv = []
+    lwlink = hass.data[LIGHTWAVE_LINK]
+
+    for device_id, device_config in discovery_info.items():
+        name = device_config[CONF_NAME]
+        serial = device_config[CONF_SERIAL]
+        trv.append(LightwaveTrv(name, device_id, lwlink, serial))
+
+    async_add_entities(trv)
+
+
+class LightwaveTrv(ClimateDevice):
+    """Representation of a LightWaveRF TRV."""
+
+    def __init__(self, name, device_id, lwlink, serial):
+        """Initialize LightwaveTrv entity."""
+        self._name = name
+        self._device_id = device_id
+        self._state = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._target_temperature_step = 0.5
+        self._hvac_mode = HVAC_MODE_HEAT
+        self._hvac_action = None
+        self._lwlink = lwlink
+        self._battery = None
+        self._serial = serial
+        # inhibit is used to prevent race condition on update.  If non zero, skip next update cycle.
+        self._inhibit = 0
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    def update(self):
+        """Communicate with a Lightwave RTF Proxy to get state."""
+        targ = temp = battery = trv_output = None
+        (temp, targ, battery, trv_output) = self._lwlink.read_trv_status(self._serial)
+        if temp is not None:
+            self._current_temperature = temp
+        if targ is not None:
+            if self._inhibit == 0:
+                self._target_temperature = targ
+                if targ == 0:
+                    # TRV off
+                    self._target_temperature = None
+                if targ >= 40:
+                    # Call for heat mode, or TRV in a fixed position
+                    self._target_temperature = None
+            else:
+                # Done the job - use proxy next iteration
+                self._inhibit = 0
+        if battery is not None:
+            self._battery = battery
+        if trv_output is not None:
+            if trv_output > 0:
+                self._hvac_action = CURRENT_HVAC_HEAT
+            else:
+                self._hvac_action = CURRENT_HVAC_OFF
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return {
+            ATTR_BATTERY_LEVEL: self._battery,
+        }
+
+    @property
+    def name(self):
+        """Lightwave trv name."""
+        return self._name
+
+    @property
+    def current_temperature(self):
+        """Property giving the current room temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Target room temperature."""
+        if self._inhibit > 0:
+            # if we get an update before the new temp has
+            # propagated, the GUI target temp is set back to the
```

Comment:
We shouldn't talk about the GUI. What we do here is to decide how the entity state is set in the state machine. The frontend GUI is just one of many consumers of that data.

Reply:
Comment changed